File: Views/Components/FreemiumPaywallCardView.swift
import SwiftUI

struct FreemiumPaywallCardView: View {
    let category: QuestionCategory
    let questionsUnlocked: Int
    let totalQuestions: Int
    let onTap: () -> Void
    
    var body: some View {
        Button(action: onTap) {
            VStack(spacing: 30) {
                Spacer()
                
                VStack(spacing: 20) {
                    // Émoji de la catégorie En couple
                    Text(category.emoji)
                        .font(.system(size: 60))
                        .padding(.bottom, 10)
                    
                    Text("congratulations".localized)
                        .font(.system(size: 32, weight: .bold))
                        .foregroundColor(.white)
                        .multilineTextAlignment(.center)
                        .padding(.horizontal, 30)
                    
                    Text("keep_going_unlock_all".localized)
                        .font(.system(size: 18))
                        .foregroundColor(.white.opacity(0.9))
                        .multilineTextAlignment(.center)
                        .padding(.horizontal, 30)
                    
                    // Bouton d'action
                    HStack(spacing: 8) {
                        Text("continue".localized)
                            .font(.system(size: 18, weight: .bold))
                            .foregroundColor(.white)
                        
                        Image(systemName: "arrow.right.circle.fill")
                            .font(.system(size: 20))
                            .foregroundColor(.white)
                    }
                    .padding(.horizontal, 24)
                    .padding(.vertical, 12)
                    .background(
                        LinearGradient(
                            gradient: Gradient(colors: [
                                Color(hex: "#FD267A"),
                                Color(hex: "#FF655B")
                            ]),
                            startPoint: .leading,
                            endPoint: .trailing
                        )
                    )
                    .cornerRadius(25)
                    .padding(.top, 20)
                }
                
                Spacer()
                
                // Logo/Branding en bas
                HStack(spacing: 8) {
                    Image("leetchi2")
                        .resizable()
                        .aspectRatio(contentMode: .fit)
                        .frame(width: 24, height: 24)
                    
                    Text("app_name".localized)
                        .font(.system(size: 16, weight: .semibold))
                        .foregroundColor(.white.opacity(0.9))
                }
                .padding(.bottom, 30)
            }
            .frame(maxWidth: .infinity, maxHeight: .infinity)
            .background(
                LinearGradient(
                    gradient: Gradient(colors: [
                        Color(red: 0.2, green: 0.1, blue: 0.15),
                        Color(red: 0.4, green: 0.2, blue: 0.3),
                        Color(red: 0.6, green: 0.3, blue: 0.2)
                    ]),
                    startPoint: .top,
                    endPoint: .bottom
                )
            )
            .frame(maxWidth: .infinity)
            .frame(height: 500)
            .cornerRadius(20)
            .overlay(
                RoundedRectangle(cornerRadius: 20)
                    .stroke(
                        LinearGradient(
                            gradient: Gradient(colors: [
                                Color(hex: "#FD267A"),
                                Color(hex: "#FF655B")
                            ]),
                            startPoint: .topLeading,
                            endPoint: .bottomTrailing
                        ),
                        lineWidth: 3
                    )
            )
            .shadow(color: .black.opacity(0.3), radius: 10, x: 0, y: 5)
        }
        .buttonStyle(PlainButtonStyle())
    }
}

#Preview {
    FreemiumPaywallCardView(
        category: QuestionCategory.categories[0],
        questionsUnlocked: 64,
        totalQuestions: 256
    ) {
        print("Paywall tapped")
    }
    .padding()
    .background(Color.black)
} 

File: Views/Components/FreemiumPaywallCardView.kt
package com.yourapp.ui.paywall

import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.rounded.ArrowForward
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.yourapp.R

data class QuestionCategory(
    val title: String,
    val emoji: String
)

@Composable
fun FreemiumPaywallCardView(
    category: QuestionCategory,
    questionsUnlocked: Int,
    totalQuestions: Int,
    onTap: () -> Unit
) {
    // Dégradés et forme (équivalents aux couleurs Swift)
    val containerShape = RoundedCornerShape(20.dp)
    val bgGradient = Brush.verticalGradient(
        colors = listOf(
            Color(red = 0.2f, green = 0.1f, blue = 0.15f),
            Color(red = 0.4f, green = 0.2f, blue = 0.3f),
            Color(red = 0.6f, green = 0.3f, blue = 0.2f)
        )
    )
    val borderGradient = Brush.linearGradient(
        colors = listOf(Color(0xFFFD267A), Color(0xFFFF655B))
    )
    val ctaGradient = Brush.linearGradient(
        colors = listOf(Color(0xFFFD267A), Color(0xFFFF655B))
    )

    Box(
        modifier = Modifier
            .fillMaxWidth()
            .height(500.dp)
            .shadow(elevation = 10.dp, shape = containerShape, clip = false)
            .clip(containerShape)
            .background(bgGradient)
            .border(width = 3.dp, brush = borderGradient, shape = containerShape)
            .clickable(onClick = onTap) // équivalent du Button avec PlainButtonStyle
            .padding(horizontal = 0.dp)
    ) {
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(horizontal = 0.dp),
            verticalArrangement = Arrangement.SpaceBetween,
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            // --- Espace haut (comme Spacer()) ---
            Spacer(modifier = Modifier.height(24.dp))

            // --- Contenu central ---
            Column(
                horizontalAlignment = Alignment.CenterHorizontally,
                modifier = Modifier.padding(horizontal = 30.dp)
            ) {
                // Emoji de la catégorie
                Text(
                    text = category.emoji,
                    fontSize = 60.sp,
                    color = Color.White,
                    modifier = Modifier.padding(bottom = 10.dp),
                    textAlign = TextAlign.Center
                )

                // Titre "congratulations"
                Text(
                    text = stringResource(id = R.string.congratulations),
                    // Équivalent context.getString(R.string.congratulations)
                    fontSize = 32.sp,
                    fontWeight = FontWeight.Bold,
                    color = Color.White,
                    textAlign = TextAlign.Center
                )

                Spacer(modifier = Modifier.height(20.dp))

                // Sous-titre "keep_going_unlock_all"
                Text(
                    text = stringResource(id = R.string.keep_going_unlock_all),
                    fontSize = 18.sp,
                    color = Color.White.copy(alpha = 0.9f),
                    textAlign = TextAlign.Center
                )

                // Bouton d'action (gradient + texte + icône)
                Box(
                    modifier = Modifier
                        .padding(top = 20.dp)
                        .clip(RoundedCornerShape(25.dp))
                        .background(ctaGradient)
                        .clickable(onClick = onTap)
                        .padding(horizontal = 24.dp, vertical = 12.dp),
                    contentAlignment = Alignment.Center
                ) {
                    Row(
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Text(
                            // ⚠️ Si ta clé s’appelle littéralement "continue", utilise l’identifiant backtick:
                            // stringResource(id = R.string.`continue`)
                            text = stringResource(id = R.string.continue_label),
                            fontSize = 18.sp,
                            fontWeight = FontWeight.Bold,
                            color = Color.White
                        )
                        Spacer(modifier = Modifier.width(8.dp))
                        Icon(
                            imageVector = Icons.Rounded.ArrowForward,
                            contentDescription = null,
                            tint = Color.White,
                            modifier = Modifier.size(20.dp)
                        )
                    }
                }
            }

            // --- Branding bas ---
            Row(
                modifier = Modifier
                    .align(Alignment.CenterHorizontally)
                    .padding(bottom = 30.dp),
                verticalAlignment = Alignment.CenterVertically
            ) {
                Image(
                    painter = painterResource(id = R.drawable.leetchi2),
                    contentDescription = null,
                    modifier = Modifier.size(24.dp),
                    contentScale = ContentScale.Fit
                )
                Spacer(modifier = Modifier.width(8.dp))
                Text(
                    text = stringResource(id = R.string.app_name),
                    fontSize = 16.sp,
                    fontWeight = FontWeight.SemiBold,
                    color = Color.White.copy(alpha = 0.9f)
                )
            }
        }
    }
}

@Preview(showBackground = true, backgroundColor = 0xFF000000)
@Composable
private fun FreemiumPaywallCardViewPreview() {
    MaterialTheme {
        FreemiumPaywallCardView(
            category = QuestionCategory(title = "En couple", emoji = "💞"),
            questionsUnlocked = 64,
            totalQuestions = 256,
            onTap = {}
        )
    }
}
